tla/ccfraft.tla

```diff
@@ -1178,8 +1178,36 @@ LogConfigurationConsistentInv ==
             /\ log[i][k].value = configurations[i][k]
             /\ log[i][k].contentType = TypeReconfiguration
 
+----
+\* Properties
+
+MonotonicTermProp ==
+    [][\A i \in Servers :
+        currentTerm[i]' >= currentTerm[i]]_vars
+
 MonotonicCommitIndexProp ==
-    [][\A i \in Servers : commitIndex[i]' >= commitIndex[i]]_vars
+    [][\A i \in Servers :
+        commitIndex[i]' >= commitIndex[i]]_vars
+
+CommittedLogNeverChangesProp ==
+    [][\A i \in Servers :
+        IsPrefix(Committed(i), Committed(i)')]_vars
+
+LeaderLogAppendOnlyProp ==
+    [][\A i \in { i \in Servers : state[i] = Leader /\ state[i]' = Leader } : 
+        IsPrefix(log[i], log[i]')]_vars
+
+CandidateLogUnchangedProp ==
+    [][\A i \in { i \in Servers : state[i] = Candidate /\ state[i]' = Candidate } : 
```

Comment:
`\in {Candidate, Pending}` or event `\in {Candidate, Pending, RetiredLeader}?

Reply:
More generalize, under what condition is a log allowed to change s.t. `log[i] # log'[i] /\ ~IsPrefix(log[i], log'[i])`?

Reply:
> `\in {Candidate, Pending}` or event `\in {Candidate, Pending, RetiredLeader}`?

Also `RetiredLeader`, and even something a bit stronger for which we don't have a state in the spec, a hypothetical `RetiredCommittableLeader` (ie. when RCI [is set](https://microsoft.github.io/CCF/main/architecture/consensus/1tx-reconfig.html#retirement-details)), which is when a retiring leader has issued a signature following its retirement transaction. The implementation does not allow a leader to issue further writes after that.

Reply:
@achamayou It would be nice to extend the spec in this way but probably out-of-scope for this PR